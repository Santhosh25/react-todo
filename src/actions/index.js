// eslint-disable-next-line import/no-anonymous-default-export
export default {
  ADD: "ADD",
  REMOVE: "REMOVE",
  CLEAR: "CLEAR",
  TOGGLE: "TOGGLE",
  ARCHIVE: "ARCHIVE",
  FILTER_ALL: "FILTER_ALL",
  EDIT: "EDIT",
  UPDATED: "UPDATED",
  FILTER_PENDING: "FILTER_PENDING",
  FILTER_COMPLETED: "FILTER_COMPLETED",
  FILTER_ARCHIVED: "FILTER_ARCHIVED",
  SEARCH: "SEARCH",
  LIST: "LIST",
  LIST_ERROR: "LIST_ERROR",
  LIST_SUCCESS: "LIST_SUCCESS"
};
