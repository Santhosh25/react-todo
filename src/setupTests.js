// jest-dom adds custom jest matchers for asserting on DOM nodes.
// allows you to do things like:
// expect(element).toHaveTextContent(/react/i)
// learn more: https://github.com/testing-library/jest-dom
import React from "react";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { configure, shallow, mount, render } from "enzyme";
import "jest-styled-components";
import "@testing-library/jest-dom";
import { Provider } from "react-redux";
import { store } from "./store";

React.useLayoutEffect = React.useEffect;
jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useLayoutEffect: jest.requireActual("react").useEffect,
}));
// const constantDate = new Date('2017-06-13T04:41:20');
// global.Date = class extends Date {
//   constructor(value) {
//     return constantDate;
//   }
// };
configure({ adapter: new Adapter() });

const wrapChildWithThemeProvider = (children) => (
  <Provider store={store}>{children}</Provider>
);

const shallowWithTheme = (children) =>
  shallow(wrapChildWithThemeProvider(children));
const mountWithTheme = (children) =>
  mount(wrapChildWithThemeProvider(children));
const renderWithTheme = (children) =>
  render(wrapChildWithThemeProvider(children));

export { shallowWithTheme, mountWithTheme, renderWithTheme };
