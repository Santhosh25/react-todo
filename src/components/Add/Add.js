import React, { useState } from "react";
import { makeStyles } from "@material-ui/core/styles";
import Paper from "@material-ui/core/Paper";
import InputBase from "@material-ui/core/InputBase";
import IconButton from "@material-ui/core/IconButton";
import Icon from "@material-ui/core/Icon";
import { useDispatch } from "react-redux";
import { v4 as uuidv4 } from "uuid";
import Tooltip from "@material-ui/core/Tooltip";

import actions from "../../actions";

const useStyles = makeStyles((theme) => ({
  root: {
    padding: "2px 15px",
    display: "flex",
    alignItems: "center",
    width: "100%",
    borderRadius: "22px",
  },
  input: {
    marginLeft: theme.spacing(1),
    flex: 1,
  },
  iconButton: {
    padding: 10,
  },
  divider: {
    height: 28,
    margin: 4,
  },
}));

const Add = () => {
  const [value, setValue] = useState("");
  const classes = useStyles();
  const dispatch = useDispatch();

  const handleSubmit = (e) => {
    e.preventDefault()
    if (value) {
      dispatch({
        type: actions.ADD,
        payload: {
          id: uuidv4(),
          text: value,
          isDone: false,
          isArchive: false,
        },
      });
    }

    setValue("");
  };

  return (
    <form noValidate onSubmit={handleSubmit}>
      <Paper className={classes.root}>
        <InputBase
          className={classes.input}
          placeholder="Add new task"
          inputProps={{ "aria-label": "add new task" }}
          value={value}
          onChange={(e) => {
            setValue(e.target.value);
          }}
        />

        <Tooltip title="Add task">
          <IconButton
            type="submit"
            className={classes.iconButton}
            aria-label="add"
          >
            <Icon>add_task</Icon>
          </IconButton>
        </Tooltip>
      </Paper>
    </form>
  );
};

export default React.memo(Add);
