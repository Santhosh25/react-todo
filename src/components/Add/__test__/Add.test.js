import React from "react";
import { v4 as uuidv4 } from "uuid";

import Add from "../Add";
import { store } from "../../../store";
import actions from "../../../actions";
import { renderWithTheme, mountWithTheme } from "../../../setupTests";

describe("Add TODO Component", () => {
  let wrapper;
  const mockFn = jest.fn();
  beforeEach(() => {
    wrapper = renderWithTheme(<Add />);
  });

  afterEach(() => {
    store.dispatch({
      type: actions.CLEAR,
    });
  });

  it("renders correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("confirm single textfield rendered", () => {
    wrapper = mountWithTheme(<Add onClick={mockFn} />);
    expect(wrapper.find("input").length).toBe(1);
  });

  it("confirm single button rendered", () => {
    wrapper = mountWithTheme(<Add onClick={mockFn} />);
    expect(wrapper.find("button").length).toBe(1);
  });

  it("add new task on button Click", () => {
    const payload = {
      id: "e98e73ed-6d3d-4fe8-96a3-ba5a990bce6d",
      text: "Tast 1",
      isDone: false,
      isArchive: false,
    };
    const expected = {
      data: [payload],
      actionType: actions.ADD,
    };
    wrapper = mountWithTheme(<Add />);
    wrapper
      .find("input")
      .at(0)
      .simulate("change", { target: { value: "Tast 1" } });
    wrapper.find("button").simulate("click");
    expect(store.getState().todos).not.toEqual(expected);
  });
});
