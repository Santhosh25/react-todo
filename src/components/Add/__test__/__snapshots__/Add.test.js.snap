// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add TODO Component renders correctly 1`] = `
LoadedCheerio {
  "0": Node {
    "attribs": Object {
      "class": "MuiPaper-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded",
    },
    "children": Array [
      Node {
        "attribs": Object {
          "class": "MuiInputBase-root makeStyles-input-2",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "aria-label": "add new task",
              "class": "MuiInputBase-input",
              "placeholder": "Add new task",
              "type": "text",
              "value": "",
            },
            "children": Array [],
            "name": "input",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "aria-label": undefined,
              "class": undefined,
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
            "x-attribsPrefix": Object {
              "aria-label": undefined,
              "class": undefined,
              "placeholder": undefined,
              "type": undefined,
              "value": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": Node {
          "attribs": Object {
            "aria-label": "add",
            "class": "MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-3",
            "tabindex": "0",
            "title": "Add task",
            "type": "button",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "class": "MuiIconButton-label",
              },
              "children": Array [
                Node {
                  "attribs": Object {
                    "aria-hidden": "true",
                    "class": "material-icons MuiIcon-root",
                  },
                  "children": Array [
                    Node {
                      "data": "add_task",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "span",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "aria-hidden": undefined,
                    "class": undefined,
                  },
                },
              ],
              "name": "span",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "class": undefined,
              },
              "x-attribsPrefix": Object {
                "class": undefined,
              },
            },
          ],
          "name": "button",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": [Circular],
          "type": "tag",
          "x-attribsNamespace": Object {
            "aria-label": undefined,
            "class": undefined,
            "tabindex": undefined,
            "title": undefined,
            "type": undefined,
          },
          "x-attribsPrefix": Object {
            "aria-label": undefined,
            "class": undefined,
            "tabindex": undefined,
            "title": undefined,
            "type": undefined,
          },
        },
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
      Node {
        "attribs": Object {
          "aria-label": "add",
          "class": "MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-3",
          "tabindex": "0",
          "title": "Add task",
          "type": "button",
        },
        "children": Array [
          Node {
            "attribs": Object {
              "class": "MuiIconButton-label",
            },
            "children": Array [
              Node {
                "attribs": Object {
                  "aria-hidden": "true",
                  "class": "material-icons MuiIcon-root",
                },
                "children": Array [
                  Node {
                    "data": "add_task",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "span",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
                "x-attribsPrefix": Object {
                  "aria-hidden": undefined,
                  "class": undefined,
                },
              },
            ],
            "name": "span",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "button",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": Node {
          "attribs": Object {
            "class": "MuiInputBase-root makeStyles-input-2",
          },
          "children": Array [
            Node {
              "attribs": Object {
                "aria-label": "add new task",
                "class": "MuiInputBase-input",
                "placeholder": "Add new task",
                "type": "text",
                "value": "",
              },
              "children": Array [],
              "name": "input",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {
                "aria-label": undefined,
                "class": undefined,
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
              "x-attribsPrefix": Object {
                "aria-label": undefined,
                "class": undefined,
                "placeholder": undefined,
                "type": undefined,
                "value": undefined,
              },
            },
          ],
          "name": "div",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": [Circular],
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {
            "class": undefined,
          },
          "x-attribsPrefix": Object {
            "class": undefined,
          },
        },
        "type": "tag",
        "x-attribsNamespace": Object {
          "aria-label": undefined,
          "class": undefined,
          "tabindex": undefined,
          "title": undefined,
          "type": undefined,
        },
        "x-attribsPrefix": Object {
          "aria-label": undefined,
          "class": undefined,
          "tabindex": undefined,
          "title": undefined,
          "type": undefined,
        },
      },
    ],
    "name": "form",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": Node {
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
    },
    "prev": null,
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
    },
  },
  "_root": LoadedCheerio {
    "0": Node {
      "children": Array [
        Node {
          "attribs": Object {},
          "children": Array [
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "head",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "body",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": [Circular],
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "parent": [Circular],
              "prev": null,
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
            Node {
              "attribs": Object {},
              "children": Array [],
              "name": "body",
              "namespace": "http://www.w3.org/1999/xhtml",
              "next": null,
              "parent": [Circular],
              "prev": Node {
                "attribs": Object {},
                "children": Array [],
                "name": "head",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": [Circular],
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {},
                "x-attribsPrefix": Object {},
              },
              "type": "tag",
              "x-attribsNamespace": Object {},
              "x-attribsPrefix": Object {},
            },
          ],
          "name": "html",
          "namespace": "http://www.w3.org/1999/xhtml",
          "next": null,
          "parent": [Circular],
          "prev": null,
          "type": "tag",
          "x-attribsNamespace": Object {},
          "x-attribsPrefix": Object {},
        },
      ],
      "name": "root",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-mode": "quirks",
    },
    "_root": [Circular],
    "length": 1,
    "options": Object {
      "decodeEntities": true,
      "xml": false,
    },
  },
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "xml": false,
  },
}
`;
